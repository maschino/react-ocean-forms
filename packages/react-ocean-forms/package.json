{
  "name": "react-ocean-forms",
  "version": "3.0.0-rc.1",
  "description": "Forms components for react based on the context api.",
  "main": "build/index.js",
  "module": "build/index.es.js",
  "typings": "build/index.d.ts",
  "private": false,
  "sideEffects": false,
  "scripts": {
    "start": "rollup -c -w",
    "build": "rollup -c",
    "start:docs": "styleguidist server",
    "build:docs": "styleguidist build",
    "build:typedocs": "typedoc"
  },
  "peerDependencies": {
    "react": "^16.8.0"
  },
  "devDependencies": {
    "react-docgen-typescript": "^1.16.3",
    "typedoc": "^0.17.3",
    "typedoc-plugin-external-module-name": "^3.0.0"
  },
  "files": [
    "build/"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/environment-agency-austria/react-ocean-forms.git"
  },
  "keywords": [
    "react",
    "reactjs",
    "forms",
    "form",
    "validation",
    "react-form",
    "react-forms",
    "react-input",
    "context"
  ],
  "author": "Denis Masinovic <denis.masinovic@umweltbundesamt.at>",
  "license": "MIT"
}
