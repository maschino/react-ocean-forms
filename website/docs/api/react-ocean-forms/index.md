---
id: "index"
title: "react-ocean-forms"
sidebar_label: "Globals"
---

