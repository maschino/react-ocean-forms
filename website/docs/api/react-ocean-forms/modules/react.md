---
id: "react"
title: "react"
sidebar_label: "react"
---

Copyright (c) 2018-present, Umweltbundesamt GmbH

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.

## Index

### Type aliases

* [PropsOf](react.md#propsof)

## Type aliases

###  PropsOf

Æ¬ **PropsOf**: *T extends function ? P : T extends object ? P : never*

Acquire the props for a Component {T}
