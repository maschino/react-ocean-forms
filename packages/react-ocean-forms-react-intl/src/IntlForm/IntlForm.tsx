/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @packageDocumentation
 * @module IntlForm
 * @category Components
 * @preferred
 */

import React, { useCallback } from 'react';

import { injectIntl } from 'react-intl';
import { Form, IMessageValues } from 'react-ocean-forms';

import { IIntlFormProps } from './IntlForm.types';

/**
 * React-intl wrapper for the OceanJS
 * Forms component.
 */
const BaseIntlForm: React.FC<IIntlFormProps> = (props) => {
  const { intl, children, ...rest } = props;

  const formatIntlString = useCallback(
    (id: string, values?: IMessageValues): string => {
      // Return the id if no valid id was given
      if (id === null || id === undefined || id === '') {
        return id;
      }

      return intl.formatMessage({ id }, values);
    },
    [intl]
  );

  return (
    <Form {...rest} formatString={formatIntlString}>
      {children}
    </Form>
  );
};

export const IntlForm = injectIntl(BaseIntlForm);
