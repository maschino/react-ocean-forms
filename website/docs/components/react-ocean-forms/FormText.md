---
title: FormText
sidebar_label: FormText
---

## Examples

### FormText

Showcase of the form text

```jsx
import { Form, FormText } from 'react-ocean-forms';

function Example() {
  return (
    <Form>
      <p>
        <FormText text="Demo output" />
      </p>

      <p>
        <FormText text="Output with parameters: {param}" values={{ param: 'Demo value' }} />
      </p>
    </Form>
  );
}

<Example />;
```

## API Reference

See [FormText API reference](../../api/react-ocean-forms/modules/formtext#const-formtext) and [prop types](../../api/react-ocean-forms/interfaces/formtext.iformtextprops).
