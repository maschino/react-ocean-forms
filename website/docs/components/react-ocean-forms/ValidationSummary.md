---
title: ValidationSummary
sidebar_label: ValidationSummary
---

## Examples

### Showcase

Showcase of the validation summary. Click on submit to display the summary.

```jsx
import { Form, Input, ValidationSummary, validators } from 'react-ocean-forms';

function Example({ logMessage }) {
  return (
    <Form className="demo">
      <ValidationSummary id="summary" />

      <Input name="input1" label="Example input 1" validators={[validators.required]} />

      <Input name="input2" label="Example input 2" validators={[validators.required]} />

      <button type="submit">Submit</button>
      <button type="reset">Reset</button>
    </Form>
  );
}

<Example />;
```

## API Reference

See [ValidationSummary API reference](../../api/react-ocean-forms/modules/validationsummary#const-validationsummary) and [prop types](../../api/react-ocean-forms/interfaces/validationsummary.ivalidationsummaryprops).
