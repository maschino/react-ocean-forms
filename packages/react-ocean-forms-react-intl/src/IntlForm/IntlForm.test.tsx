import React from 'react';

import { IntlProvider } from 'react-intl';
import { FormText } from 'react-ocean-forms';

import { render } from '@testing-library/react';

import { IntlForm } from './IntlForm';

describe('<IntlForm />', () => {
  it('should render without crashing', () => {
    const { asFragment, getByText } = render(
      <IntlProvider locale="en">
        <IntlForm>mock</IntlForm>
      </IntlProvider>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText('mock')).toBeVisible();
  });

  it('should pass the correct message translation method to the base form', () => {
    const { getByText, queryByText } = render(
      <IntlProvider locale="en" messages={{ test: 'translated text' }}>
        <IntlForm>
          <FormText text="test" />
        </IntlForm>
      </IntlProvider>
    );

    expect(queryByText('test')).toBeNull();
    expect(getByText('translated text')).toBeVisible();
  });

  it('should pass the message params as well', () => {
    const { getByText, queryByText } = render(
      <IntlProvider locale="en" messages={{ test: 'translated text {foo}' }}>
        <IntlForm>
          <FormText text="test" values={{ foo: 'bar' }} />
        </IntlForm>
      </IntlProvider>
    );

    expect(queryByText('test')).toBeNull();
    expect(getByText('translated text bar')).toBeVisible();
  });

  describe('error cases', () => {
    const cases: [string, null | undefined | string][] = [
      ['null', null],
      ['undefined', undefined],
      ['empty', ''],
    ];

    describe.each(cases)('called with "%s"', (name, value) => {
      it('should return the passed value as-is', () => {
        const { getByTestId } = render(
          <IntlProvider locale="en">
            <IntlForm>
              <div data-testid="result">
                {/*
                 // TODO: Replace with expected error when new typescript version drops
                 // @ts-ignore */}
                <FormText text={value} />
              </div>
            </IntlForm>
          </IntlProvider>
        );

        expect(getByTestId('result')).toBeEmpty();
      });
    });
  });
});
