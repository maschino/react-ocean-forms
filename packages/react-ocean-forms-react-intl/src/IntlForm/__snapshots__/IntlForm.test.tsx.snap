// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntlForm /> should render without crashing 1`] = `
<Component
  formatString={[Function]}
/>
`;
